store/src/java/com/zimbra/cs/service/util/S3BucketGlobalConfigUtil.java

```diff
@@ -0,0 +1,485 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.service.util;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.GlobalExternalStoreConfigConstants;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.ldap.LdapUtil;
+import com.zimbra.soap.admin.message.CreateS3BucketConfigRequest;
+import com.zimbra.soap.admin.message.DeleteS3BucketConfigRequest;
+import com.zimbra.soap.admin.type.Attr;
+import com.zimbra.soap.type.GlobalExternalStoreConfig;
+import com.zimbra.soap.type.GlobalS3BucketConfiguration;
+
+public class S3BucketGlobalConfigUtil {
+
+	/**
+     * Parse the json which comes from the GlobalExternalStoreConfig LDAP parameter to object.
+     *
+     * @param gescLdapJson
+     * @return GlobalExternalStoreConfig
+     * @throws ServiceException
+     */
+    public static GlobalExternalStoreConfig parseJsonToGesc(String gescLdapJson) throws ServiceException {
+    	
+    	GlobalExternalStoreConfig gesc = new GlobalExternalStoreConfig();
+    	
+    	try {
+    		
+    		if(gescLdapJson != null && !gescLdapJson.isEmpty()) {
```

Comment:
indentation issue all over file and add space after if

Reply:
done!